src/pydvl/value/shapley/actor.py

```diff
@@ -179,71 +124,48 @@ def __init__(
         :param u: Utility object with model, data, and scoring function
         :param coordinator: worker results will be pushed to this coordinator
         :param worker_id: id used for reporting through maybe_progress
-        :param progress: Whether to display a progres bar
         :param update_period: interval in seconds between different updates to
             and from the coordinator
-
+        :param truncation: callable that decides whether to stop computing
+            marginals for a given permutation.
         """
         super().__init__(
-            coordinator=coordinator,
-            update_period=update_period,
-            worker_id=worker_id,
-            progress=progress,
+            coordinator=coordinator, update_period=update_period, worker_id=worker_id
         )
         self.u = u
-        self.num_samples = len(self.u.data)
-        self.pbar = maybe_progress(
-            self.num_samples,
-            self.progress,
-            position=worker_id,
-            desc=f"Worker {worker_id}",
-        )
-        self._iteration_count = 1
-        self._avg_values: Optional[Union[float, "NDArray"]] = None
-        self._var_values: Optional[Union[float, "NDArray"]] = None
+        self.truncation = truncation
 
-    def _compute_values(self, *args, **kwargs) -> "NDArray":
-        # Import here to avoid errors with circular imports
-        from .montecarlo import _permutation_montecarlo_marginals
+    def _compute_marginals(self) -> ValuationResult:
+        # Avoid circular imports
+        from .montecarlo import _permutation_montecarlo_shapley
 
-        return _permutation_montecarlo_marginals(self.u, max_permutations=1)[0]  # type: ignore
+        return _permutation_montecarlo_shapley(
+            self.u,
+            done=MaxChecks(1),
```

Comment:
MaxChecks is a special policy that increments a counter on each call to check(), instead of looking at number of updates and so on.